lufopyramids.sh

```diff
@@ -16,81 +23,205 @@
+# gdal_retile.py -v -r cubic -levels 5 -resume -ps 8192 8192 -co TILED=YES -co COMPRESS=JPEG -co PHOTOMETRIC=YCBCR -s_srs "EPSG:28992" -targetDir $DESTDIR $DESTDIR_ROOT/overview.vrt
 
-#if [ -d $DESTDIR ] ; then
-#    rm -rf $DESTDIR
-#fi
+### Put original resolution images in their own subdirectory
+mv $DESTDIR/*.tif $DESTDIR/0
 
-mkdir -p $DESTDIR/0
-gdal_retile.py -v -r cubic -levels 5 -ps 8192 8192 -co TILED=YES -co COMPRESS=JPEG -co PHOTOMETRIC=YCBCR -s_srs "EPSG:28992" -targetDir $DESTDIR --optfile /tmp/list.txt
+### Set Levels and loop over them
+### This exists to ensure public WMS usage is much quicker as the TIF files are in the blob storage
+LEVELS=$(find $DESTDIR/* -type d -exec basename {} \; | sort)
+for i in $LEVELS; do
+
+### Create overviews to show when zoomed out of original resolution
+    for j in $DESTDIR/$i/*.tif; do
+        gdaladdo -r cubic --config COMPRESS_OVERVIEW JPEG --config PHOTOMETRIC_OVERVIEW YCBCR --config INTERLEAVE_OVERVIEW PIXEL $j 2 4 8 16 32 64 128
+    done
 
-# Put original resolution images in their own subdirectory
+### Create tile indices for Mapserver WMS usage
+    gdaltindex $DESTDIR/imagery-$YEAR-$i.shp $DESTDIR/$i/*.tif
+    ogrinfo -dialect SQLITE -sql "UPDATE 'imagery-$YEAR-$i' SET location = '/vsicurl/https://files.data.amsterdam.nl/$TYPE/$YEAR/pyramid' || SUBSTR(location,length('$DESTDIR/'),length(location))" $DESTDIR/imagery-$YEAR-$i.shp
+    ogrinfo -sql "CREATE SPATIAL INDEX ON imagery-$YEAR-$i" $DESTDIR/imagery-$YEAR-$i.shp
+
+done
+
+### Create tile index for Mapserver Tiling only
+gdaltindex $DESTDIR_ROOT/imagery.shp $DESTDIR/0/*.tif
+
+# # -----------------------------------------
+# # Documentation:
+# # -----------------------------------------
+# #
+# # http://www.gdal.org/frmt_gtiff.html
+# # https://smathermather.wordpress.com/2016/05/01/efficient-delivery-of-raster-data-part-4/
+# # https://smathermather.wordpress.com/2016/04/15/whichever-tiler-you-use-and-efficient-delivery-of-raster-data-image-pyramid-layer-update2/
+# # https://astuntech.atlassian.net/wiki/display/ISHAREHELP/Mosaic+thousands+of+raster+images
+# # http://blog.cleverelephant.ca/2015/02/geotiff-compression-for-dummies.html
+# # https://www.slidesearch.net/slide/geoserver-in-production-we-do-it-here-is-how-foss4g-2016
+# # https://www.mapbox.com/blog/super-sharp-pleiades-imagery-on-mapbox/
+# # https://github.com/boundlessgeo/workshops/blob/master/workshops/data_configs/sphinx/source/raster.rst
+# # http://www.ianturton.com/tutorials/bluemarble.html
+# #
+# # -----------------------------------------
+
+# # find . -type f -size 1651c -delete
+# # find . -type d -empty -delete
+# # for i in {24..25}
+# # do
+# #   for j in {15..24}
+# #   do
+# #       cp -v Ortho_1090_4917_${j}_${i}.tif ..
+# #   done
+# # done
+
+# # for f in Ortho*; do mv $f ${f/Ortho/0_Ortho}; done
```

Comment:
Seems you've added your worksheet below: mostly same code as above...?

Reply:
Right, that can be removed